assets/controllers/materialize/form_select_controller.js

```diff
@@ -0,0 +1,12 @@
+import { Controller } from '@hotwired/stimulus';
+import M from '@materializecss/materialize';
+
+export default class extends Controller {
+  connect() {
+    this.instance = M.FormSelect.init(this.element.querySelector('select'));
```

Comment:
Stimulus controller applied on a parent element and get the select element to pass it to Materialize because of the magic Materialize is doing with the select element, it is altered by Materialize and made Stimulus do an infinite loop when applied directly on the select element